src/poetry/console/commands/self/show/plugins.py

```diff
@@ -8,15 +8,32 @@
 
 
 if TYPE_CHECKING:
-    from entrypoints import EntryPoint
     from poetry.core.packages.package import Package
 
+    from poetry.utils._compat import metadata
+
 
 @dataclasses.dataclass
 class PluginPackage:
     package: Package
-    plugins: list[EntryPoint] = dataclasses.field(default_factory=list)
-    application_plugins: list[EntryPoint] = dataclasses.field(default_factory=list)
+    plugins: list[metadata.EntryPoint] = dataclasses.field(default_factory=list)
+    application_plugins: list[metadata.EntryPoint] = dataclasses.field(
+        default_factory=list
+    )
+
+    def append(self, entry_point: metadata.EntryPoint) -> None:
+        from poetry.plugins.application_plugin import ApplicationPlugin
+        from poetry.plugins.plugin import Plugin
```

Comment:
any particular reason for these imports to be here and not at the top of the file?

Reply:
it is mostly done to defer the import so the load times are quicker; for this change I am simply maintaining the status quo. Any changes to import ordering should be dealt with outside of this PR i reckon.